query/src/test/scala/filodb/query/exec/rangefn/LabelFunctionSpec.scala

```diff
@@ -0,0 +1,114 @@
+package filodb.query.exec.rangefn
+
+import com.typesafe.config.{Config, ConfigFactory}
+import monix.execution.Scheduler.Implicits.global
+import monix.reactive.Observable
+import org.scalatest.{FunSpec, Matchers}
+import org.scalatest.concurrent.ScalaFutures
+
+import filodb.core.MetricsTestData
+import filodb.core.query.{CustomRangeVectorKey, RangeVector, RangeVectorKey, ResultSchema}
+import filodb.memory.format.{RowReader, ZeroCopyUTF8String}
+import filodb.query._
+import filodb.query.exec.TransientRow
+
+class LabelFunctionSpec extends FunSpec with Matchers with ScalaFutures {
+
+  val config: Config = ConfigFactory.load("application_test.conf").getConfig("filodb")
+  val resultSchema = ResultSchema(MetricsTestData.timeseriesDataset.infosFromIDs(0 to 1), 1)
+  val ignoreKey = CustomRangeVectorKey(
+    Map(ZeroCopyUTF8String("ignore") -> ZeroCopyUTF8String("ignore")))
+  val queryConfig = new QueryConfig(config.getConfig("query"))
+
+  it("should  replace label only when match is found in label replace") {
+    val sampleKey1 = CustomRangeVectorKey(
+      Map(ZeroCopyUTF8String("instance") -> ZeroCopyUTF8String("demo.io:9090"),
+        ZeroCopyUTF8String("job") -> ZeroCopyUTF8String("test")))
+    val sampleKey2 = CustomRangeVectorKey(
+      Map(ZeroCopyUTF8String("instance") -> ZeroCopyUTF8String("demo.io:8080")))
+
+    val sampleWithKey: Array[RangeVector] = Array(
+      new RangeVector {
+        override def key: RangeVectorKey = sampleKey1
+
+        override def rows: Iterator[RowReader] = Seq(
+          new TransientRow(1L, 3.3d),
+          new TransientRow(2L, 5.1d)).iterator
+      },
+      new RangeVector {
+        override def key: RangeVectorKey = sampleKey2
+
+        override def rows: Iterator[RowReader] = Seq(
+          new TransientRow(3L, 100d),
+          new TransientRow(4L, 200d)).iterator
+      })
+
+    val expectedLabels = List(Map(ZeroCopyUTF8String("instance") -> ZeroCopyUTF8String("demo.io new Label Value 90"),
+      ZeroCopyUTF8String("job") -> ZeroCopyUTF8String("test")),
+      Map(ZeroCopyUTF8String("instance") -> ZeroCopyUTF8String("demo.io:8080"))) // will not be replaced
+
+    val funcParams = Seq("instance", "$1 new Label Value $2", "instance", "(.*):90(.*)")
```

Comment:
There is one more test case... What if there are just two groups and I use $3 ?

Reply:
Added it in test case "should works with multiple groups and remove groups which do exist from the replacement string"